parser/src/ast/boundary_constraints.rs

```diff
@@ -7,7 +7,32 @@ use std::fmt::Display;
 /// Stores the boundary constraints to be enforced on the trace column values.
 #[derive(Debug, PartialEq)]
 pub struct BoundaryConstraints {
-    pub boundary_constraints: Vec<BoundaryConstraint>,
+    variables: Vec<BoundaryVariable>,
+    boundary_constraints: Vec<BoundaryConstraint>,
+}
+
+impl BoundaryConstraints {
+    /// Creates a new instance of [BoundaryConstraints] sith the specified variables and
+    /// boundary constraints
+    pub fn new(
+        variables: Vec<BoundaryVariable>,
+        boundary_constraints: Vec<BoundaryConstraint>,
```

Comment:
You are right, in order to be able to do that we'll probably need to preserve these as statements in the parser so that we can check for this in the IR. I'll address in the next PR that adds variables to IR. Please let me know if I should fix this in this PR itself.

Reply:
Fixing in the next PR is fine, but I wonder if fixing it here could be pretty simple. For example, maybe we just have a single vector would work? Something like:
```Rust
pub struct BoundaryConstraints {
    statements: Vec<BoundaryStmt>,
}
```

Reply:
Agreed. This is what I had in mind as well. I'll handle it in this PR itself. If we go with this route, another question here would be that if there is only one struct field, maybe we can get rid of the `BoundaryConstraints` and `TransitionConstraints` struct wrappers and just store the sections as `BoundaryConstraints(Vec<BoundaryStmt>)` and `TransitionConstraints(Vec<TransitionStmt>)` instead.

Reply:
yep, that should be fine.

Reply:
Done!